Client/Frontend/Browser/TabManager.swift

```diff
@@ -209,8 +209,9 @@ class TabManager: NSObject {
 
     //Called by other classes to signal that they are entering/exiting private mode
     //This is called by TabTrayVC when the private mode button is pressed and BEFORE we've switched to the new mode
-    func willSwitchTabMode() {
-        if shouldClearPrivateTabs() && (selectedTab?.isPrivate ?? false) {
+    //The force flag is used to make sure that when exiting PBM without selecting a private tab we still remove tabs
+    func willSwitchTabMode(force: Bool = false) {
+        if shouldClearPrivateTabs() && ((selectedTab?.isPrivate ?? false) || force)  {
```

Comment:
Why does it matter whether or not the selected tab is private? Can we just remove that check?